sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -518,7 +533,40 @@ def refresh(self, **kwargs) -> None:
             elif need_refresh and self._on_refresh_success:
                 self._on_refresh_success()
 
+    def _check_configuration_settings(self, key, label, etag, headers, **kwargs) -> Tuple[bool, Union[ConfigurationSetting,  None]]:
+        try:
+            updated_sentinel = self._client.get_configuration_setting(  # type: ignore
+                key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, headers=headers, **kwargs
+            )
+            if updated_sentinel is not None:
+                logging.debug(
+                    "Refresh all triggered by key: %s label %s.",
+                    key,
+                    label,
+                )
+                return True, updated_sentinel
+        except HttpResponseError as e:
+            if e.status_code == 404:
+                if etag is not None:
+                    # If the sentinel is not found, it means the key/label was deleted, so we should refresh
+                    logging.debug("Refresh all triggered by key: %s label %s.", key, label)
+                    return True, updated_sentinel
```

Comment:
In the event of a 404, will `updated_sentinel` be initialized? Should we just return `None` to be explicit?